File: #0523.continuous-subarray-sum.cpp
class Solution {
public:
    bool checkSubarraySum(vector<int>& nums, int k) {

        int size = nums.size();
        if (size == 1) {
            return false;
        }

        unordered_map<int,int> m;
        int sum[size];
        sum[0] = nums[0];
        m[sum[0] % k] = 0;
        for (int i = 1; i < size; i++) {
            sum[i] = (sum[i - 1] + nums[i]);
            if (sum[i] % k == 0) {
                return true;
            }
            
            if (m.count(sum[i] % k) && m[sum[i] % k] < i - 1) {
                return true;
            }
            
            if (!m.count(sum[i] % k)) {
                m[sum[i] % k] = i;
            }
        }
        return false;
    }
};

File: #0523.continuous-subarray-sum.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {boolean}
 */
var checkSubarraySum = function(nums, k) {
	const sums = { 0: -1 };
	let sum = 0;
	for (let i = 0; i < nums.length; i++) {
		sum += nums[i];
		sum %= k;
		if (sums[sum] === undefined) {
            sums[sum] = i;
        } else if (i - sums[sum] > 1) {
            return true;
        }
	}
	return false;
};
